freeciv-web/src/main/webapp/javascript/player.js

```diff
@@ -247,3 +247,27 @@ function player_has_wonder(playerno, improvement_id)
   }
   return false;
 }
+
+/**************************************************************************
+  Checks if a username is valid.
+  Returns a textual reason for invalid names, null for valid ones.
+**************************************************************************/
+function get_invalid_username_reason(username)
+{
+  if (username == null || username.length == 0) {
+    return "empty";
+  } else if (username.length <= 2 ) {
+    return "too short";
+  } else if (username.length >= 32) {
+    return "too long";
+  }
+  username = username.toLowerCase();
+  if (username == "pbem") {
+    return "invalid";
```

Comment:
Maybe "not available" would be more clear to a user seeing the message (minor personal pref for the second.).